openlibrary/tests/solr/test_update.py

```diff
@@ -60,8 +62,8 @@ def make_work(**kw):
 class FakeDataProvider(DataProvider):
     """Stub data_provider and methods which are used by build_data."""
 
-    docs: list = []
-    docs_by_key: dict = {}
+    docs: tuple = ()
+    docs_by_key = MappingProxyType({})
```

Comment:
Not sure why, but this one appears to be necessary.
```suggestion
    docs_by_key: MappingProxyType = MappingProxyType({})
```